### PR TITLE
refactor: roundtrip test logic - #407

### LINKED ISSUE #407: Refactor: Tests
# Discussion 🗣

The code written for tests in `markdown-docx`  is repeated in many places. Refactor the code so that it follows the `DRY` principle.

## Context

In `OoxmlTransformer.test.js` and `CiceroMarkToOOXMLRoundTrip.test.js` the code logic is repeated.

### PR DESCRIPTION
**Pull request by K-Kumar-01**
Signed-off-by: k-kumar-01 <kushalkumargupta4@gmail.com>

# Closes #407<CORRESPONDING ISSUE NUMBER>

Refactors the test logic in `CiceroMarkToOOXMLRoundTrip.test.js`

### Changes

- <ONE> Use a function to check for the equality of ciceroMark(s).
- <TWO> Replace the logic with the function.

### Author Checklist
- [x] Ensure you provide a [DCO sign-off](https://github.com/probot/dco#how-it-works) for your commits using the `--signoff` option of git commit.
- [x] Vital features and changes captured in unit and/or integration tests
- [x] Commits messages follow [AP format](https://github.com/accordproject/techdocs/blob/master/DEVELOPERS.md#commit-message-format)
- [ ] Extend the documentation, if necessary
- [ ] Merging to `master` from `fork:branchname`

### CODE REVIEW

**Review by algomaster99 — CHANGES_REQUESTED** [review]
This is a good start. But I think it can be refactored further. For example, you're still hardcoding the filename of your test data. Maybe you can avoid that. What you can probably do is use `fs` module to iterate over the files in the directory and run your test function over them. Once you do this, you won't have to edit your test files a lot and you will be done just by adding a test case for each node you write the transformations.

**Comment by K-Kumar-01** [comment]
@algomaster99 
I have updated the PR. Let me know if there are any changes required.

**Review by algomaster99 — CHANGES_REQUESTED** [review]
Minor change. Then we are good to go.

**Comment by K-Kumar-01** [comment]
@algomaster99 
I am unable to see the changes requested. Can you please mention them in a comment?

File: packages/markdown-docx/test/CiceroMarkToOOXMLRoundTrip.test.js [hunk 1]
```diff
@@ -17,43 +17,20 @@
...
-
-        const ooxmlTransformer = new OoxmlTransformer();
-        const convertedObject = ooxmlTransformer.toCiceroMark(ooxml);
-        expect(convertedObject).to.deep.equal(textandHeadingCiceroMark);
-    });
+    const fileNames = fs.readdirSync(directoryName);
+
+    for (const file of fileNames) {
+        // acceptance-of-delivery requires advance transformer. Skip for now.
+        if (!file.startsWith('acceptance')) {
```
Comment (algomaster99) [inline packages/markdown-docx/test/CiceroMarkToOOXMLRoundTrip.test.js:30]: Hardcode the exact filename instead. `startswith` implies that all files that starts with `acceptance` should be skipped.

**Review by algomaster99 — CHANGES_REQUESTED** [review]
Here you go.

File: packages/markdown-docx/test/CiceroMarkToOOXMLRoundTrip.test.js (hunk 1 above)
Comment (K-Kumar-01) [reply packages/markdown-docx/test/CiceroMarkToOOXMLRoundTrip.test.js:30]: @algomaster99 Since we will be removing this later, do we need to change this?

Comment (algomaster99) [reply packages/markdown-docx/test/CiceroMarkToOOXMLRoundTrip.test.js:30]: Commit histories are often referred to. You don't want to disseminate false information.

Comment (K-Kumar-01) [reply packages/markdown-docx/test/CiceroMarkToOOXMLRoundTrip.test.js]: @algomaster99 
Pushed the latest change.

**Review by algomaster99 — APPROVED** [review]

**Pull request merged**